### DISCUSSION TITLE
Problems with permissions on PROD

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#16856] [Archive]
Hello,

It would appear that I have all sorts of problems with permissions when accessing the APP on production.
I am only able to set these permissions when accessing the APP in dev environment (localhost:8055), because logging in as admin on production doesn't work. But it shouldn't matter... permissions should be carried over right? Anyway, this is what I'm dealing with:

1 - Admin access doesn't work on production environment (related issue: https://github.com/directus/directus/discussions/16359)
![image](https://user-images.githubusercontent.com/56647386/208481366-f3bce93f-39b4-4e03-aa17-2db5909c26a6.png)

2 - File Upload seems to not work for non-admin role (even though all possible permissions are set -- see screenshot below):
![image](https://user-images.githubusercontent.com/56647386/208480477-175fbd4c-2dfd-4fc0-a275-b94b3778770d.png)

3 - All sort of GET errors 
![image](https://user-images.githubusercontent.com/56647386/208481306-4c33e231-592a-4ff6-bd3c-1879b2cbd593.png)

Here is a screenshot of the current REVIEWER role permissions that I have set just to test the simple actions like the ones above:
![image](https://user-images.githubusercontent.com/56647386/208480532-f183fa75-378e-47ca-8354-0afffcb737ba.png)

My environment is Windows Server (IIS 7)
Directus 9.20.4

Let me know if you need any other detail

**Comment by AlexVanK** [comment]
Update:

It seems that properly setting the PUBLIC_URL variable in the .ENV file does address problem 2 and 3 above.
Still having trouble connecting to PROD as admin:

![image](https://user-images.githubusercontent.com/56647386/208483044-6ec21d82-08be-4a5b-96d6-b72b76430b98.png)

**Comment by br41nslug** [comment]
I doubt this is an actual permission problem 😬 Permission denied is also the security fallback so non-existing items or invalid URLs (by proxies for instance) will also result in this.

Can you set Directus `LOG_LEVEL` to `debug` and find the error that is thrown with the 500 request?

**Comment by rijkvanzanten** [comment]
ECONNRESET also sounds like wherever you're running Directus has trouble connecting to / keeping a connection to the database.

**Reply by AlexVanK** [reply]
I don't have that problem when connecting as non-admin.
The connection to the DB fails only when authenticating as admin

**Comment by AlexVanK** [comment]
LOG_LEVEL debug:

![image](https://user-images.githubusercontent.com/56647386/208490113-3d091736-e819-4902-bc9c-55bd6cc2ce2e.png)
![image](https://user-images.githubusercontent.com/56647386/208490172-e71b5a27-10fe-4c7d-a1a5-80b5add35a23.png)

**Reply by br41nslug** [reply]
Sounds like you're missing OS permissions for `uv_uptime`. this should only be a problem for the `/server/info` endpoint i think as it is explicitly requested there using `process.uptime()`.

**Reply by AlexVanK** [reply]
Thank you @br41nslug! 
Are you familiar with IIS setup: application/application pools, or where on Windows Server I should address `uv_uptime` permissions?
![image](https://user-images.githubusercontent.com/56647386/208492484-afbbd441-261a-4f27-b899-7dcb8e0ec223.png)

Anyway, why would this issue affect only admin users?

**Reply by br41nslug** [reply]
Im am not familiar with that im afraid (only run windows on desktop beyond that linux all the way). The reason why this only happens for admins is because this logic is only run for admins (non-admins don't get system information on this endpoint).

**Reply by AlexVanK** [reply]
Wait, still a bit confused... Why would this affect only production environment and not dev? The Machine is the same

**Reply by br41nslug** [reply]
Ah hmmm thats strange indeed. I had assumed those would have been separate machines

**Reply by RosilBelli** [reply]
> Thank you @br41nslug! Are you familiar with IIS setup: application/application pools, or where on Windows Server I should address `uv_uptime` permissions? ![image](https://user-images.githubusercontent.com/56647386/208492484-afbbd441-261a-4f27-b899-7dcb8e0ec223.png)
> 
> Anyway, why would this issue affect only admin users?

Hi @AlexVanK!

Did you get directus to work on IIS. This is also my case and my problem!
I could not find a solution yet

**Reply by AlexVanK** [reply]
Hi @RosilBelli,

My Directus instance works on IIS. Development AND Production, the only problem is that I can't log into production using an Admin account... everything else works.
